File: Lab-4/recursion.c
#include<stdio.h>
long int tailFact(int n,long int a)
{
    if(n==1)
        return a;
    else
        return tailFact(n-1, a*n);
}

int main()
{
    int n;
    printf("enter the number:");
    scanf("%d",&n);
    printf("The factorial of %d is:%ld",n,tailFact(n,1));
    return 0;
}

File: Lab-4/recursion.py
def tail_fact(n, a):
    if n == 1:
        return a
    else:
        return tail_fact(n - 1, a * n)

n = int(input("Enter the number: "))
print(f"The factorial of {n} is: {tail_fact(n, 1)}")
